FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,47 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case .success((let data, let httpResponse)):
+				completion(FeedLoaderDecoder.decode(with: data, httpResponse: httpResponse))
+			}
+		}
+	}
+}
+
+private extension RemoteFeedLoader {
+	private struct FeedLoaderDecoder {
+		private static let OK_200 = 200
+		private struct RootRemoteResponse: Decodable {
+			let items: [FeedImageItem]
+		}
+
+		private struct FeedImageItem: Decodable {
+			private let image_id: UUID
+			private let image_desc: String?
+			private let image_loc: String?
+			private let image_url: URL
+
+			var item: FeedImage {
+				FeedImage(id: image_id, description: image_desc, location: image_loc, url: image_url)
+			}
+		}
+
+		static func decode(with data: Data, httpResponse: HTTPURLResponse) -> FeedLoader.Result {
+			if httpResponse.statusCode != FeedLoaderDecoder.OK_200 {
+				return FeedLoader.Result.failure(Error.invalidData)
+			}
+			let decoder = JSONDecoder()
+			if let feedLoader = try? decoder.decode(RootRemoteResponse.self, from: data) {
+				return FeedLoader.Result.success(feedLoader.items.map {
```

Comment:
There's no need to write `FeedLoader.Result.success`, the Type System can infer that for us based on the return type for this function, so we can make the code more readable:
```suggestion
				return .success(feedLoader.items.map {
```
The same applies for the `.failure` case.